### PR TITLE
Chartmuseum svc account

### PR DESCRIPTION
**Pull request by eviln1**
This PR adds support for ServiceAccount creation / use and for pod securityContext; none will be set by default.
On clusters with PSP enabled and a bit tightened, this chart might fail since `system:serviceaccount:{{ namespace }}:default` will likely be unable to run a container as root (image default)

### CODE REVIEW

File: stable/chartmuseum/templates/deployment.yaml [hunk 1]
```diff
@@ -89,7 +89,13 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
     {{- end }}
-
+    {{- if or .Values.serviceAccount.name .Values.serviceAccount.create }}
+      serviceAccountName: {{ .Values.serviceAccount.name | default (include "chartmuseum.serviceAccountName" .) }}
```
Comment (jdolitsky) [inline stable/chartmuseum/templates/deployment.yaml]: Ok, so I think that if you are creating a service account via the chart, we should not allow you to change its name. I think this check needs to look something more like
```
{{- if .Values.serviceAccount.create }}
serviceAccountName: {{ include "chartmuseum.fullname" . }}
{{- else if .Values.serviceAccount.name }}
serviceAccountName: {{ .Values.serviceAccount.name }}
{{- end }}
```

File: stable/chartmuseum/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.serviceAccount.create -}}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "chartmuseum.serviceAccountName" . }}
```
Comment (jdolitsky) [inline stable/chartmuseum/templates/serviceaccount.yaml]: lets just use `chartmuseum.fullname` here

File: stable/chartmuseum/templates/_helpers.tpl [hunk 3]
```diff
@@ -81,4 +81,15 @@ chartmuseum-0.4.5
...
 {{- define "chartmuseum.chartref" -}}
 {{- replace "+" "_" .Chart.Version | printf "%s-%s" .Chart.Name -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "chartmuseum.serviceAccountName" -}}
```
Comment (jdolitsky) [inline stable/chartmuseum/templates/_helpers.tpl]: this helper should be removed, not needed based on comments above

**Review by jdolitsky — CHANGES_REQUESTED** [review]
Thanks for submitting! Overall looks great, just some comments around the naming of the service account

**Comment by eviln1** [comment]
Amended per your request.

**Pull request merged**